mantidproject/mantid
# Update IN5 pixel shape, monitor detID

The pixel radius and height for ILL's IN5 instrument were slightly off. This PR fixes the radius to take the 0.5mm thick tube walls into account and gives the pixels more accurate height. Also, monitor's detector ID is changed from 0 to 100000 to follow the convention of IN6.

**Report to:** [nobody].

**To test:**

Make sure Mantid is not picking up a wrong IDF from the cache. In Linux, check the `~/.mantid/instrument` directory.

Check that the new pixel radius corresponds to the tube diameter (25.4 - 1)mm and that the height is 2950mm / 241.

To check that the pixel positions have not changed except for the more accurate coordinate values used in the new IDF
1. Grab the old IDF from `master` or download from GitHub
2. Replace `path/to/old/IDF` below with the path to the old IDF
3. Run the script. The script should not print anything, just finish silently
```python
def within_delta(a, b, delta):
    return abs(a - b) / a < delta

in5_old = LoadEmptyInstrument(Filename='path/to/old/IDF/IN5_Defintion.xml')
in5_new = LoadEmptyInstrument(InstrumentName='IN5')
specInfo_old = in5_old.spectrumInfo()
specInfo_new = in5_new.spectrumInfo()
assert(specInfo_old.size() == specInfo_new.size())
assert(specInfo_old.l1() == specInfo_new.l1())
for i in range(specInfo_old.size() - 1):
    assert(specInfo_old.isMonitor(i + 1) == specInfo_new.isMonitor(i))
    twoTheta1 = specInfo_old.twoTheta(i + 1)
    twoTheta2 = specInfo_new.twoTheta(i)
    if not within_delta(twoTheta1, twoTheta2, 1e-5):
        print('2theta mismatch at spectrum {}, 1: {} 2: {}'.format(i, twoTheta1, twoTheta2))
        break
    l2_old = specInfo_old.l2(i + 1)
    l2_new = specInfo_new.l2(i)
    if not within_delta(l2_old, l2_new, 1e-7):
        print('l2 mismatch at spectrum {}, 1: {} 2: {}'.format(i, l2_old, l2_new))
        break
```

Fixes #24137.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
